### ISSUE TITLE
5.1.0 Error on Azure Devops ubuntu Agents

### ISSUE THREAD

**Issue by MattMencel** [GitTools/GitVersion#1872]
I just noticed my builds on AzDO have started failing with the new 5.1.0 release. I've tried with the Ubuntu 16.04 and 18.04 agents.

```
- script: |
    dotnet tool install --global GitVersion.Tool
    ~/.dotnet/tools/dotnet-gitversion /output buildserver
  displayName: "GitVersion Setup and Run"
```

```
  ERROR [10/28/19 17:05:54:37] An unexpected error occurred:
LibGit2Sharp.LibGit2SharpException: remote authentication required but no callback set
   at LibGit2Sharp.Core.Ensure.HandleError(Int32 result)
   at LibGit2Sharp.Core.Ensure.ZeroResult(Int32 result)
   at LibGit2Sharp.Core.Proxy.git_remote_fetch(RemoteHandle remote, IEnumerable`1 refSpecs, GitFetchOptions fetchOptions, String logMessage)
   at LibGit2Sharp.Commands.Fetch(Repository repository, String remote, IEnumerable`1 refspecs, FetchOptions options, String logMessage)
   at GitVersion.Helpers.GitRepositoryHelper.Fetch(ILog log, AuthenticationInfo authentication, Remote remote, Repository repo) in D:\a\1\s\src\GitVersionCore\Helpers\GitRepositoryHelper.cs:line 145
   at GitVersion.Helpers.GitRepositoryHelper.NormalizeGitDirectory(ILog log, IEnvironment environment, String gitDirectory, AuthenticationInfo authentication, Boolean noFetch, String currentBranch, Boolean isDynamicRepository) in D:\a\1\s\src\GitVersionCore\Helpers\GitRepositoryHelper.cs:line 40
   at GitVersion.GitPreparer.Prepare(Boolean normalizeGitDirectory, String currentBranch, Boolean shouldCleanUpRemotes) in D:\a\1\s\src\GitVersionCore\GitPreparer.cs:line 52
   at GitVersion.GitVersionCalculator.CalculateVersionVariables() in D:\a\1\s\src\GitVersionCore\GitVersionCalculator.cs:line 52
   at GitVersion.ExecCommand.Execute() in D:\a\1\s\src\GitVersionExe\ExecCommand.cs:line 40
   at GitVersion.GitVersionExecutor.VerifyArgumentsAndRun(Arguments arguments) in D:\a\1\s\src\GitVersionExe\GitVersionExecutor.cs:line 117
 ```

**Comment by arturcic** [comment]
can you try the hotfix version, 5.1.1?

**Comment by MattMencel** [comment]
Tried 5.1.1 and seems it's the same issue?  For the moment I can stay on 5.0.1, but if you want more information just let me know.

```

INFO [10/29/19 14:09:16:35] Begin: Normalizing git directory for branch 'refs/heads/my-branch'
  INFO [10/29/19 14:09:16:40] One remote found (origin -> 'https://FOO.visualstudio.com/PROJECT/_git/BAR').
  INFO [10/29/19 14:09:16:41] Fetching from remote 'origin' using the following refspecs: +refs/heads/*:refs/remotes/origin/*.
  INFO [10/29/19 14:09:16:48] End: Normalizing git directory for branch 'refs/heads/my-branch' (Took: 125.34ms)
  ERROR [10/29/19 14:09:16:50] An unexpected error occurred:
LibGit2Sharp.LibGit2SharpException: remote authentication required but no callback set
   at LibGit2Sharp.Core.Ensure.HandleError(Int32 result)
   at LibGit2Sharp.Core.Proxy.git_remote_fetch(RemoteHandle remote, IEnumerable`1 refSpecs, GitFetchOptions fetchOptions, String logMessage)
   at LibGit2Sharp.Commands.Fetch(Repository repository, String remote, IEnumerable`1 refspecs, FetchOptions options, String logMessage)
   at GitVersion.Helpers.GitRepositoryHelper.NormalizeGitDirectory(ILog log, IEnvironment environment, String gitDirectory, AuthenticationInfo authentication, Boolean noFetch, String currentBranch, Boolean isDynamicRepository) in D:\a\1\s\src\GitVersionCore\Helpers\GitRepositoryHelper.cs:line 40
   at GitVersion.GitPreparer.Prepare(Boolean normalizeGitDirectory, String currentBranch, Boolean shouldCleanUpRemotes) in D:\a\1\s\src\GitVersionCore\GitPreparer.cs:line 51
   at GitVersion.GitVersionCalculator.CalculateVersionVariables() in D:\a\1\s\src\GitVersionCore\GitVersionCalculator.cs:line 52
   at GitVersion.ExecCommand.Execute() in D:\a\1\s\src\GitVersionExe\ExecCommand.cs:line 40
   at GitVersion.GitVersionExecutor.VerifyArgumentsAndRun(Arguments arguments) in D:\a\1\s\src\GitVersionExe\GitVersionExecutor.cs:line 117
  INFO [10/29/19 14:09:16:50] 
  INFO [10/29/19 14:09:16:50] Attempting to show the current git graph (please include in issue): 
  INFO [10/29/19 14:09:16:50] Showing max of 100 commits
INFO [10/29/19 14:09:16:31] Applicable build agent found: 'AzurePipelines'.INFO [10/29/19 14:09:16:32] Working directory: /home/vsts/work/1/sINFO [10/29/19 14:09:16:32] IsDynamicGitRepository: FalseINFO [10/29/19 14:09:16:34] Returning Project Root from DotGitDirectory: /home/vsts/work/1/s/.git/ - /home/vsts/work/1/s/INFO [10/29/19 14:09:16:35] Running on Unix.INFO [10/29/19 14:09:16:35] Applicable build agent found: 'AzurePipelines'.INFO [10/29/19 14:09:16:35] Branch from build environment: refs/heads/my-branchINFO [10/29/19 14:09:16:35] Begin: Normalizing git directory for branch 'refs/heads/my-branch'  INFO [10/29/19 14:09:16:40] One remote found (origin -> 'https://FOO.visualstudio.com/PROJECT/_git/BAR').  INFO [10/29/19 14:09:16:41] Fetching from remote 'origin' using the following refspecs: +refs/heads/*:refs/remotes/origin/*.  INFO [10/29/19 14:09:16:48] End: Normalizing git directory for branch 'refs/heads/my-branch' (Took: 125.34ms)  ERROR [10/29/19 14:09:16:50] An unexpected error occurred:
LibGit2Sharp.LibGit2SharpException: remote authentication required but no callback set
   at LibGit2Sharp.Core.Ensure.HandleError(Int32 result)
   at LibGit2Sharp.Core.Proxy.git_remote_fetch(RemoteHandle remote, IEnumerable`1 refSpecs, GitFetchOptions fetchOptions, String logMessage)
   at LibGit2Sharp.Commands.Fetch(Repository repository, String remote, IEnumerable`1 refspecs, FetchOptions options, String logMessage)
   at GitVersion.Helpers.GitRepositoryHelper.NormalizeGitDirectory(ILog log, IEnvironment environment, String gitDirectory, AuthenticationInfo authentication, Boolean noFetch, String currentBranch, Boolean isDynamicRepository) in D:\a\1\s\src\GitVersionCore\Helpers\GitRepositoryHelper.cs:line 40
   at GitVersion.GitPreparer.Prepare(Boolean normalizeGitDirectory, String currentBranch, Boolean shouldCleanUpRemotes) in D:\a\1\s\src\GitVersionCore\GitPreparer.cs:line 51
   at GitVersion.GitVersionCalculator.CalculateVersionVariables() in D:\a\1\s\src\GitVersionCore\GitVersionCalculator.cs:line 52
   at GitVersion.ExecCommand.Execute() in D:\a\1\s\src\GitVersionExe\ExecCommand.cs:line 40
```

**Comment by arturcic** [comment]
@MattMencel, can you try the version 5.1.2-beta.1+4 and let me know if it fixed the error?

**Comment by arturcic** [comment]
@MattMencel any updates?

**Comment by MattMencel** [comment]
Hey @arturcic sorry about the delay... No errors! 👍

**Comment by arturcic** [comment]
Closed by #1879
